### ISSUE TITLE
[android] AnimatedLayout, view's second childs are not getting removed properly on conditional rendering

### ISSUE THREAD

**Issue by hannojg** [software-mansion/react-native-reanimated#2247]
## Description

Subviews, that are contained in an `Animated.View` which is contained in an `AnimatedLayout` behave buggy on android, see this video:

https://user-images.githubusercontent.com/16821682/127782115-0ae70e42-f509-43c8-b83e-ff805b0759b4.mp4

### Expected behavior

The second view should also be removed

### Actual behavior & steps to reproduce

The second view doesn't get removed.

## Snack or minimal code example

```jsx
export default function App() {
  const [isVisible, setIsVisible] = useState(false);

  return (
    <AnimatedLayout style={styles.container}>
      <Animated.View entering={SlideInRight} exiting={SlideOutRight}>
        <Text>REA AnimatedLayout android bug reproduction</Text>
        <Button title="Toggle" onPress={() => setIsVisible((prev) => !prev)} />
        {isVisible && (<View>
          <Text>I am the first view.</Text>
          <Text>I am the second, and I won't get removed :(</Text>
        </View>)}
      </Animated.View>
    </AnimatedLayout>
  );
}
```

A full reproduction is here: https://github.com/hannojg/rea-android-reproduction

The issue above can be fixed by using an `Animated.View` instead of just a `View` in the conditional rendering. 
Or by just not wrapping everything in an `AnimatedLayout` 😄

## Package versions

- React Native: 0.64.2
- React Native Reanimated: 2.3.0-alpha2 (also happening on alpha1)
- NodeJS: 15
- Xcode: /
- Java & Gradle: 1.8.0_241 (Oracle Corporation 25.241-b07), Gradle 6.8

## Affected platforms

- [x] Android
- [ ] iOS
- [ ] Web

**Comment by hannojg** [comment]
The issue doesn't appear anymore when you wrap the view that is rendering the error in an `Animated.View` 🤔

**Comment by jmysliv** [comment]
Hi @hannojg ! We have just released `2.3.0-alpha.3` without `AnimatedLayout` and I think your problem might be gone. Can you check it and let me know?

**Comment by jmysliv** [comment]
Hi @hannojg, I am closing this issue due to inactivity. If you still encounter this issue, let me know, and I'll reopen it.
